libsolidity/analysis/NameAndTypeResolver.cpp

```diff
@@ -456,7 +455,11 @@ bool DeclarationRegistrationHelper::registerDeclaration(
 
 	string name = _name ? *_name : _declaration.name();
 	Declaration const* shadowedDeclaration = nullptr;
-	if (_warnOnShadow && !name.empty() && _container.enclosingContainer())
+	// Do not warn about shadowing for structs and enums because their members are
+	// not accessible without prefixes. Also do not warn about event parameters
+	// because they don't participate in any proper scope.
```

Comment:
```suggestion
	// because they do not participate in any proper scope.
```
I think we have a rule in the guideline not to use contractions.

Reply:
Done. There are quote some `n't`s in our code base :).